docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,205 @@
+|`text`       |carries a human readable event-text; must be UTF-8 encoded|
+|`timestamp`  |optional time that indicates when the event has occurred; when not provided, thin-edge.io uses the current system time as the time of the event; when provided must be conform to ISO 8601|
+
+TODO: add somehow "an event can optionally contain additional custom information"
+
+  * **behaviour of an event:**
+    - thin-edge does not store any historical occurrences for events
+
+### Alarms
+**alarms** are notifications about some critical behaviour of the device's environment or software system;
+e.g. when a temperature sensor detects a temperature went out of its valid range
+
+```javascript
+{
+    // example for an alarm
+    "text": "Temperature is very high",  // 'text' message of that alarm
+    "time": "2021-01-01T05:30:45+00:00"  // optional 'timestamp' of that alarm
+}
+```
+
+|Reference  |Description|
+| --------- | --------- |
+|`type_name`  |a string that identifies the alarm uniquely in context of the device; it is not part of the alarm's data elements, instead it is part of the MQTT topics when the alarm message is sent to thin-edge.io|
+|`severity`   |a string that indicates the severity of the alarm; must be `critical`, `major`, `minor` or `warning`; it is not part of the alarm's data elements, instead it is part of the MQTT topics when the alarm message is sent to thin-edge.io |
+|`text`       |carries a human readable alarm-text; must be UTF-8 encoded|
+|`timestamp`  |optional time that indicates when the alarm has occurred; when not provided, thin-edge.io uses the current system time as the time of the alarm; when provided must be conform to ISO 8601|
+
+TODO: add somehow "an alarm can optionally contain additional custom information"
+
+  * **behaviour of an alarm:**
+    - thin-edge does not store any historical occurrences for alarms
+    - **alarms** are stateful; i.e. once raised, an **alarm** is active until it was explicitly cleared by the device's software or the cloud
+
+[^1]: details see "Domain Model" appendix [Device Domain](./domain-model.md#device-overview) -->
+
+## Use of MQTT
+
+**thin-edge** expects the MQTT broker [mosquitto](https://mosquitto.org/) to be available on the device.
+**thin-edge** uses **mosquitto** to consume and provide telemetry data. All telemetry data are reflected with specific MQTT topics and payload in JSON format.
+
+**thin-edge** assumes **mosquitto** is configured in a secure manner, to avoid any inappropriate access to **thin-edge** topics and payload.
+Any malicious access to the broker can hazard **thin-edge** and all connected devices. Mosquitto provides a wide range of authentication and access control options. For more details see _Authentication_ and _ACL_ (Access Control List) in [mosquitto documentation](https://mosquitto.org/man/mosquitto-conf-5.html).
+
+### Telemetry Data
```

Comment:
I'm not sure if this section is really required now that the example of each element is shown in the section at the top of the page.

What if we move the missing details (e.g. topic name, retain etc.) from this section to the top? Or does that couple the model too much to MQTT?

Reply:
Sections "Telemetry Data" and "Use of MQTT" are reordered and partially merged now:
- put section 'Use of MQTT' before section 'Telemetry Data'
- merge telemetry specifics from section 'MQTT' into 'Telemetry Data'

Details see https://github.com/thin-edge/thin-edge.io/pull/1677/commits/0795970fa016ac62b302d18066a351c715037e70